java/solomiia_tymoshchuk/src/com/zoolatech/lecture4/tasks/_4/Unit.java

```diff
@@ -0,0 +1,93 @@
+package com.zoolatech.lecture4.tasks._4;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+public enum Unit {
+    GRAM("g", new Aliases("gram", "grams", "Gram", "gramms")),
+    KILOGRAM("kg", new Aliases("kilos", "kilo", "Kilograms", "kgs", "kilograms")),
+    OUNCE("oz", new Aliases("ounce", "ounces")),
+    POUND("lb", new Aliases("Pound", "lbs", "pound")),
+    TON("t", new Aliases("ton", "Ton", "tonnes", "tns"));
+
+
+    private final String abbreviation;
+    private final Unit.Aliases aliases;
+
+    Unit(String abbreviation, Aliases aliases) {
+        this.abbreviation = abbreviation;
+        this.aliases = aliases;
+    }
+
+    public String getAbbreviation() {
+        return abbreviation;
+    }
+
+    public static Unit getByAlias(String alias) {
+        for (Unit unit : Unit.values()) {
+            if (unit.hasAlias(alias.toLowerCase())) return unit;
+        }
+        return null;
+    }
+
+    public boolean hasAlias(String al) {
+        if (aliases == null) return false;
+        else return aliases.contains(al);
+    }
+
+    static double convertValue(double value, Unit necessaryValue, Unit initialUnit) {
```

Comment:
1. the method shouldn't be static
2. the method should accept two arguments: value and initialUnit (source unit)
3. necessaryValue is current object (this)